source/delegate/contracts/Delegate.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.23;
+
+import "./interfaces/IDelegate.sol";
+import "@airswap/swap-erc20/contracts/interfaces/ISwapERC20.sol";
+import { ERC20 } from "solady/src/tokens/ERC20.sol";
+import { Ownable } from "solady/src/auth/Ownable.sol";
+import { SafeTransferLib } from "solady/src/utils/SafeTransferLib.sol";
+
+/**
+ * @title Delegate: Deployable Trading Rules for the AirSwap Network
+ * @notice Supports fungible tokens (ERC-20)
+ * @dev inherits IDelegate, Ownable and uses SafeTransferLib
+ */
+contract Delegate is IDelegate, Ownable {
+  // The Swap contract to be used to settle trades
+  ISwapERC20 public swapERC20;
+
+  // Mapping of sender to senderToken to to signerToken to Rule
+  mapping(address => mapping(address => mapping(address => Rule))) public rules;
+
+  // Mapping of signer to authorized signatory
+  mapping(address => address) public authorized;
+
+  /**
+   * @notice Contract Constructor
+   */
+  constructor(ISwapERC20 _swapERC20) {
+    swapERC20 = _swapERC20;
```

Comment:
Probably also want a setter for swapERC20 similar to Wrapper.

Reply:
Done